mu.h

```diff
@@ -133,8 +144,18 @@ class MuCorrProvider : public CorrectionsBase<MuCorrProvider> {
             muIDCorrections["NUM_TightID_DEN_TrackerMuons"]=corrections_->at("NUM_TightID_DEN_TrackerMuons");
             muIDCorrections["NUM_LoosePFIso_DEN_TightID"]=corrections_->at("NUM_LoosePFIso_DEN_TightID");
             muIDCorrections["NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight"]=corrections_->at("NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight");
+
+            muIDCorrections["NUM_MediumID_DEN_TrackerMuons"]=corrections_->at("NUM_MediumID_DEN_TrackerMuons");
+            muIDCorrections["NUM_LoosePFIso_DEN_MediumID"]=corrections_->at("NUM_LoosePFIso_DEN_MediumID");
+            muIDCorrections["NUM_IsoMu24_DEN_CutBasedIdMedium_and_PFIsoMedium"]=corrections_->at("NUM_IsoMu24_DEN_CutBasedIdMedium_and_PFIsoMedium");
+
         }
     }
+    float getMuonSFMedium(const LorentzVectorM & muon_p4, const float Muon_pfRelIso04_all, const bool Muon_mediumID, UncSource source, UncScale scale) const {
```

Comment:
No need to create an additional method. Get Muon ID WP from the config and pass it to the constructor.